timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -316,11 +317,104 @@ void declare_restraint(py::module &m, const char *typestr) {
         py::array_t<double, py::array::c_style> buffer({B, 3});
         grad.get_du_dp_tangents(buffer.mutable_data());
         return buffer;
-    });;
+    });
 
 }
 
 
+template <typename RealType>
+void declare_centroid_restraint(py::module &m, const char *typestr) {
+
+    using Class = timemachine::CentroidRestraint<RealType>;
+    std::string pyclass_name = std::string("CentroidRestraint_") + typestr;
+    py::class_<Class, timemachine::Gradient>(
+        m,
+        pyclass_name.c_str(),
+        py::buffer_protocol(),
+        py::dynamic_attr()
+    )
+    .def(py::init([](
+        const py::array_t<int, py::array::c_style> &group_a_idxs,
+        const py::array_t<int, py::array::c_style> &group_b_idxs,
+        const py::array_t<double, py::array::c_style> &masses,
+        double kb,
+        double b0,
+        int lambda_flag,
+        int lambda_offset
+    ){
+        std::vector<int> vec_group_a_idxs(group_a_idxs.size());
```

Comment:
it's hard to beat 
```
std::vector<int> vec_group_a_idxs = group_a_idxs;
```

Reply:
I'm also confused why these need to be copied at all, CentroidRestraint takes const refs

Reply:
Ah, the inputs aren't vectors here, they are 
```
const py::array_t<int, py::array::c_style>
```

Reply:
You can do tree the numpy array as a pointer and do:

```
std::vector<int> vec_group_a_idxs(group_a_idxs.data(),
                                                       group_a_idxs.data() + group_a_idx.data().size());
```

Reply:
Yeah I think later on we definitely need to clean this up. I was thinking of adding Eigen as dependency into the C++ later (which pybind11 has many excellent adapters for). Though I do like the alternate constructor to avoid having to call memcpy explicitly.